FlowCrypt/Functionality/Services/Key Services/PassPhraseStorage.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  PassPhraseStorageService.swift
+//  FlowCrypt
+//
+//  Created by Anton Kharchevskyi on 02.06.2021.
+//  Copyright © 2021 FlowCrypt Limited. All rights reserved.
+//
+
+import UIKit
+
+protocol PassPhraseStorageType {
+    func getPassPhrases() -> [PassPhrase]
+    func savePassPhrase(with passPhrase: PassPhrase, inStorage: Bool)
+    func updatePassPhrase(with passPhrase: PassPhrase, inStorage: Bool)
+}
+
+protocol EmailProviderType {
+    var email: String? { get }
+}
+
+final class PassPhraseStorage: PassPhraseStorageType {
+    private lazy var logger = Logger.nested(Self.self)
+
+    let currentUserEmail: String?
+    let storage: EncryptedPassPhraseStorage
+    let localStorage: LocalPassPhraseStorageType
+    let timeoutContext: (component: Calendar.Component, timeout: Int)
+
+    /// used for tests only, otherwise seconds will be used
+    let isHours: Bool
```

Comment:
Better would be to input amount of seconds total. Then in tests you can input `1`, but in prod it's `4 * 3600`.